### ISSUE TITLE
Behavior when persisting files

### ISSUE THREAD

**Issue by karaolidis** [nix-community/impermanence#197]
The `README` says: "If the file exists in persistent storage, it will be bind mounted to the target path; otherwise it will be symlinked."

Is there a specific reason for this behavior that I am missing out on? In my use case, some files spot the broken symlink, delete it, and replace it with their own file that ends up not being persisted. Wouldn't it make more sense to do:

```bash
...

if [[ -L "$mountPoint" && $(readlink -f "$mountPoint") == "$targetFile" ]]; then
    trace "$mountPoint already links to $targetFile, ignoring"
elif mount | grep -F "$mountPoint"' ' >/dev/null && ! mount | grep -F "$mountPoint"/ >/dev/null; then
    trace "mount already exists at $mountPoint, ignoring"
elif [[ -e "$mountPoint" ]]; then
    echo "A file already exists at $mountPoint!" >&2
    exit 1
fi

if [[ ! -e "$targetFile" ]]; then
    touch "$targetFile"
fi

touch "$mountPoint"
mount -o bind "$targetFile" "$mountPoint"
```

Instead of:

```bash
...

if [[ -L "$mountPoint" && $(readlink -f "$mountPoint") == "$targetFile" ]]; then
    trace "$mountPoint already links to $targetFile, ignoring"
elif mount | grep -F "$mountPoint"' ' >/dev/null && ! mount | grep -F "$mountPoint"/ >/dev/null; then
    trace "mount already exists at $mountPoint, ignoring"
elif [[ -e "$mountPoint" ]]; then
    echo "A file already exists at $mountPoint!" >&2
    exit 1
elif [[ -e "$targetFile" ]]; then
    touch "$mountPoint"
    mount -o bind "$targetFile" "$mountPoint"
else
    ln -s "$targetFile" "$mountPoint"
fi
```

I can open a PR if so :)

Thanks.

**Comment by karaolidis** [comment]
Just had a read through #1, I think adding an option to specify the behavior when a file doesn't exist might be a good call?

**Comment by nazarewk** [comment]
I'm actually hitting an issue related to this behavior: a lot of software seems to replace symlink with raw files preventing those from being stored at all.

I could really use an option to specify the default content of the file if it doesn't exist in either place so it is ALWAYS bind-mounted.

Another option would be to create a oneshot `.service` triggered by a [`PathExists=`](https://www.freedesktop.org/software/systemd/man/latest/systemd.path.html#PathExists=) to move the file into `/persist` and create bind mount as soon as it appears for the first time.

**Comment by prettymuchbryce** [comment]
I agree with @nazarewk. It seems like it should be possible for the existing `persist-persist-x.service` oneshot service to move the existing file into `/persist` on the first run if currently exists. Maybe there is some constraint I'm not aware of that makes this undesirable, but the current experience of having to do this manually is a little kludgy. Currently you either manually move the file to `/persist`, or you rename it first so it doesn't collide, and then rename it again after applying the config. I have only seen this happen with files myself, so I am not sure if the same issue occurs with directories as well.
